yarn-project/txe/src/oracle/txe_oracle.ts

```diff
@@ -221,11 +219,36 @@ export class TXE implements TypedOracle {
     return this.txeDatabase.addAuthWitness(authWitness.requestHash, authWitness.witness);
   }
 
-  async addNullifiers(contractAddress: AztecAddress, nullifiers: Fr[]) {
+  async addPublicDataWrites(writes: PublicDataUpdateRequest[]) {
     const db = await this.trees.getLatest();
-    const siloedNullifiers = nullifiers.map(nullifier => siloNullifier(contractAddress, nullifier).toBuffer());
 
-    await db.batchInsert(MerkleTreeId.NULLIFIER_TREE, siloedNullifiers, NULLIFIER_SUBTREE_HEIGHT);
+    // only insert the last write to a given slot
+    const uniqueWritesSet = new Map<bigint, PublicDataUpdateRequest>();
+    for (const write of writes) {
+      uniqueWritesSet.set(write.leafSlot.toBigInt(), write);
+    }
+    const uniqueWrites = Array.from(uniqueWritesSet.values());
+
+    await db.batchInsert(
+      MerkleTreeId.PUBLIC_DATA_TREE,
+      uniqueWrites.map(w => new PublicDataTreeLeaf(w.leafSlot, w.newValue).toBuffer()),
+      0,
+    );
+  }
```

Comment:
public executor no longer commits to worldStateDB, so we need to manually perform tree insertions for public data writes here.